src/form.js

```diff
@@ -898,14 +837,27 @@ define([
                 mug.options.afterInsert(this, mug);
             }
         },
-        _updateMugPath: function (mug, oldPath, newPath) {
-            var map = this.mugMap;
-            delete map[oldPath];
-            if (_.isUndefined(newPath)) {
+        _updateMugPath: function (mug, oldHashtag, newHashtag) {
+            var map = this.mugMap, newPath;
+            delete map[this.normalizeHashtag(oldHashtag)];
+            if (oldHashtag) {
+                this.removeHashtag(oldHashtag);
+            }
+            if (_.isUndefined(newHashtag)) {
                 newPath = mug.absolutePath;
+                newHashtag = mug.hashtagPath;
+            } else {
+                newPath = newHashtag.replace(/^#form/, '/' + this.tree.rootNode.rootNodeId);
```

Comment:
would it make sense to use `this.getBasePath(true)` instead of `'/' + this.tree.rootNode.rootNodeId`?